src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -821,7 +834,16 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
     config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
 
     # Retrieving Helm chart OCI Artifact location
-    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+    if container_registry_repository:
+        if auto_upgrade != "false":
+            auto_upgrade_error = 'Disable auto upgrade when using custom repository'
+            telemetry.set_exception(exception=auto_upgrade_error, fault_type=consts.Custom_Repository_Disable_Auto_Upgrade_Fault_Type, summary=auto_upgrade_error)
+            raise ValidationError(auto_upgrade_error, recommendation='use --auto_upgrade false')
+        registry_path = "{}/azurearck8s/azure-arc-k8sagents:1.3.8".format(container_registry_repository)
```

Comment:
same comment as above, we need to remove the agent version hardcoding and parametrize that

Reply:
ack